types/index.d.ts

```diff
@@ -184,6 +184,14 @@ export class Rule implements RuleProperties {
   ): T extends true ? string : RuleSerializable;
 }
 
+interface BooleanConditionResultProperties {
+  result: boolean
+}
+
+interface ConditionResultProperties extends BooleanConditionResultProperties {
+  factResult: any
```

Comment:
generally we should start using `unknown` for values like this rather than `any`

Reply:
done!